packages/fiori/src/FlexibleColumnLayout.ts

```diff
@@ -0,0 +1,815 @@
+			this.fireLayoutChange(false, true);
+		}
+	}
+
+	startArrowClick() {
+		this.arrowClick({ start: true, end: false });
+	}
+
+	endArrowClick() {
+		this.arrowClick({ start: false, end: true });
+	}
+
+	arrowClick(options: { start: boolean, end: boolean }) {
+		// update public property
+		this.layout = this.nextLayout(this.layout, { start: options.start, end: options.end })!;
+
+		// update layout
+		this.updateLayout();
+
+		// fire layout-change
+		this.fireLayoutChange(true, false);
+	}
+
+	updateLayout() {
+		this._width = this.widthDOM;
+		this._columnLayout = this.nextColumnLayout(this.layout);
+		this._visibleColumns = this.calcVisibleColumns(this._columnLayout);
+		this.toggleColumns();
+	}
+
+	syncLayout() {
+		if (this._prevLayout !== this.layout) {
+			this.updateLayout();
+			this._prevLayout = this.layout;
+		}
+	}
+
+	toggleColumns() {
+		this.toggleColumn("start");
+		this.toggleColumn("mid");
+		this.toggleColumn("end");
+	}
+
+	toggleColumn(column: string) {
+		let columnWidth;
+		let columnDOM;
+
+		if (column === "start") {
+			columnWidth = this.startColumnWidth;
+			columnDOM = this.startColumnDOM;
+		} else if (column === "mid") {
+			columnWidth = this.midColumnWidth;
+			columnDOM = this.midColumnDOM;
+		} else {
+			columnWidth = this.endColumnWidth;
+			columnDOM = this.endColumnDOM;
+		}
+
+		const currentlyHidden = columnWidth === "0px";
+		const previouslyHidden = columnDOM.style.width === "0px";
+
+		// no change
+		if (currentlyHidden && previouslyHidden) {
+			return;
+		}
+
+		// column resizing: from 33% to 67%, from 25% to 50%, etc.
+		if (!currentlyHidden && !previouslyHidden) {
+			columnDOM.style.width = columnWidth;
+			return;
+		}
+
+		// hide column: 33% to 0, 25% to 0, etc .
+		if (currentlyHidden) {
+			// animate the width
+			columnDOM.style.width = columnWidth;
+
+			// hide column with delay to allow the animation runs entirely
+			columnDOM.addEventListener("transitionend", this.columnResizeHandler);
+
+			return;
+		}
+
+		// show column: from 0 to 33%, from 0 to 25%, etc.
+		if (previouslyHidden) {
+			columnDOM.removeEventListener("transitionend", this.columnResizeHandler);
+			columnDOM.classList.remove("ui5-fcl-column--hidden");
+			columnDOM.style.width = columnWidth;
+		}
+	}
+
+	columnResizeHandler = (e: Event) => {
+		(e.target as HTMLElement).classList.add("ui5-fcl-column--hidden");
+	}
+
+	nextLayout(layout: FCLLayout, arrowsInfo: { start?: boolean, end?: boolean }) {
+		if (!arrowsInfo) {
+			return;
+		}
+
+		if (arrowsInfo.start) {
+			return getNextLayoutByStartArrow()[layout as keyof typeof getNextLayoutByStartArrow];
+		}
+
+		if (arrowsInfo.end) {
+			return getNextLayoutByEndArrow()[layout as keyof typeof getNextLayoutByEndArrow];
+		}
+	}
+
+	nextColumnLayout(layout: FCLLayout) {
+		return this._effectiveLayoutsByMedia[this.media][layout].layout;
+	}
+
+	calcVisibleColumns(colLayot: Array<string>) {
+		return colLayot.filter(col => col !== null && col !== undefined).length;
```

Comment:
previously we were checking if the column width is 0. instead of checking for null and undefined make the filtration base on column width in px ( col !== "0px")